src/adapter/graphql/src/queries/flows/flow_outcome.rs

```diff
@@ -7,16 +7,19 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
+use kamu_core::DatasetRepository;
 use kamu_flow_system::FlowError;
 
 use crate::prelude::*;
+use crate::queries::{Account, Dataset};
 
 ///////////////////////////////////////////////////////////////////////////////
 
 #[derive(Union)]
 pub(crate) enum FlowOutcome {
     Success(FlowSuccessResult),
     Failed(FlowFailedError),
+    DatasetCompacted(FlowDatasetCompactedFailedError),
```

Comment:
It's not a new type of outcome, it's still `Failed`. We want to have union in `FlowFailedError` instead.

Reply:
Changed